include/kwk/detail/sequence/shape_constraints.hpp

```diff
@@ -12,6 +12,8 @@
 
 #if !defined(KWK_DEFAULT_SHAPE_CONSTRAINTS)
 #define KWK_DEFAULT_SHAPE_CONSTRAINTS ::kwk::basic_shape_checks
+#elif defined(KWK_DEFAULT_SHAPE_CONSTRAINTS_INCLUDE)
+#include KWK_DEFAULT_SHAPE_CONSTRAINTS_INCLUDE
```

Comment:
what is the intent here ?

Reply:
this starts to be a lot :D

Reply:
> this starts to be a lot :D

but this actually works now :)
your shape thingy converts and constructs according to my rules, specified in my sources 🎉 :)

Reply:
OK then. But can't we end up with just requiring `KWK_DEFAULT_SHAPE_CONSTRAINTS_INCLUDE` and ask for this file to contain the #define ? So the logic becomes try to incldue the include and if it fails, set up the default case ?